### PR TITLE
feat: disable nav bar items for not onboarded users with incomplete profile

### PR DESCRIPTION
**Pull request by ana-yldio**
https://trello.com/c/Udf2NtaX/1302-disable-nav-bar-items-for-not-onboarded-users-with-incomplete-profile
## Requirements
- [x] main navbar items are disabled and show no interaction
- [x] there is no link cursor in the ASAP logo
- [x] on the user's nav bar "My team" button should be disabled

### CODE REVIEW

**Review by kmaid — CHANGES_REQUESTED** [review]

**Review by migcarva — APPROVED** [review]
lgtm

**Review by jhonny-m — APPROVED** [review]

File: packages/react-components/src/organisms/__tests__/UserNavigation.test.tsx [hunk 1]
```diff
@@ -42,3 +43,17 @@ it('applies the passed href', () => {
     ),
   ).toHaveAttribute('href', '/profile');
 });
+
+it('enables My team link when user is onboarded', () => {
+  const { getAllByText, rerender } = render(
+    <UserNavigation {...props} userOnboarded={false} />,
```
Comment (kmaid) [inline packages/react-components/src/organisms/__tests__/UserNavigation.test.tsx:49]: You should put the team within this test. We try to use the generic props to meet the minimum conditions and everything that is asserted on is overwritten within the test to try and avoid changing tests in future.

File: packages/react-components/src/organisms/UserNavigation.tsx [hunk 2]
```diff
@@ -43,11 +43,13 @@ const bottomLinksStyles = css({
 });
 
 export interface UserNavigationProps {
+  readonly userOnboarded: boolean;
```
Comment (kmaid) [inline packages/react-components/src/organisms/UserNavigation.tsx:46]: This should be optional and default to true

**Review by kmaid — CHANGES_REQUESTED** [review]

File: packages/react-components/src/organisms/__tests__/UserNavigation.test.tsx (hunk 1 above)
Comment (kmaid) [reply packages/react-components/src/organisms/__tests__/UserNavigation.test.tsx:49]:
```suggestion
    <UserNavigation {...props} teams={[{ name: 'Team 1', href: '/team-1' }]} userOnboarded={false} />,
```

**Pull request closed without merging**